tests/plugins/dns_list_base.js

```diff
@@ -211,8 +208,7 @@ function zone_disable_test_func (zones, test, cb) {
     const again = () => {
         i++;
         setTimeout(() => {
-            if (this.plugin.zones.length === zones.length) return fin_check();
-            if (i > 4) return fin_check();
+            if (this.plugin.zones.length === zones.length || i > 4) return fin_check();
```

Comment:
I'm not a fan of replacing two very simple statements with one more complicated one.